File: smartmall-backend/src/main/java/com/smartmall/service/ProductService.java
package com.smartmall.service;

import com.smartmall.entity.Product;

import java.util.List;

public interface ProductService {
	Product create(Product product);

	List<Product> getAll();

	List<Product> getByShopId(Long shopId);

	Product getById(Long id);

	Product update(Long id, Product product);

	void delete(Long id);
}

File: smartmall-frontend/src/services/productService.js
import api from "./api";

export const productService = {
	list: (shopId) => api.get("/products", { params: shopId ? { shopId } : {} }),
	get: (id) => api.get(`/products/${id}`),
	create: (payload) => api.post("/products", payload),
	update: (id, payload) => api.put(`/products/${id}`, payload),
	remove: (id) => api.delete(`/products/${id}`),
};

export default productService;
